Kernel/Process.cpp

```diff
@@ -3041,14 +3065,10 @@ void Process::die()
     if (m_tracer)
         m_tracer->set_dead();
 
-    {
-        // Tell the threads to unwind and die.
-        InterruptDisabler disabler;
-        for_each_thread([](Thread& thread) {
-            thread.set_should_die();
-            return IterationDecision::Continue;
-        });
-    }
+    Thread::current->set_should_die();
+
+    // Tell the other threads to unwind and die.
+    kill_threads_except_self();
```

Comment:
`Process::die()` may be called on a not-currently-running process during signal dispatch.
This logic would not work in that case, since this assumes that `Thread::current` is a thread owned by `this`.

Reply:
Indeed, this was a mistake. I solved in, now we just kill all the process threads, we don't use `Thread::current` any more.